.github/workflows/dotnet.yml

```diff
@@ -17,9 +17,14 @@ jobs:
       uses: actions/setup-dotnet@v1
       with:
         dotnet-version: 6.0.x
-    - name: Restore dependencies
-      run: dotnet restore
-    - name: Build
-      run: dotnet build --no-restore
-    - name: Test
-      run: dotnet test --no-build --verbosity normal
+    - name: Restore packages, build and test
+      run: dotnet test --verbosity normal -c Release -o ./output
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - name: Publish
+      if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+      run: nuget push ./output/Duffel.ApiClient.*.nupkg -ApiKey ${{secrets.NUGET_API_KEY
```

Comment:
Have you tested this step yet – if so, how? 😸

Reply:
I tagged a commit locally, ran '[act](https://github.com/nektos/act)' made sure the publish job ran and tried to publish, didn't see it succeed in publishing because I didn't provide a secret token and the error was ```error: Missing value for option 'api-key'``` which I think is close enough. I could spend some more time figuring out how to test pass secrets to act and test that way but not sure it is worth the effort vs trying it and fixing what fails